finality-aleph/src/network/gossip/service.rs

```diff
@@ -104,6 +106,34 @@ impl<D: Data, P: Clone + Debug + Eq + Hash + Send + 'static> Network<D> for Serv
     }
 }
 
+struct NamedFuturesBoundedSender<T> {
+    name: &'static str,
+    tx: mpsc::Sender<T>,
+}
+
+impl<T> NamedFuturesBoundedSender<T> {
+    pub fn send(&mut self, msg: T) -> Result<(), TrySendError<T>> {
+        self.tx.try_send(msg).map_err(|e| {
+            if e.is_full() {
+                warn!(
```

Comment:
Logging should rather happen outside, next to the other `if` you changed. In particular then you cannot (but also don't have to) use the name – from context you will know which one it is, and even better, to whom the dropped message was addressed. Since you don't have to do that you can just use a plain `Sender`, so you don't even need this struct, just construct a `mpsc::channel` and you are done.

tl;dr You put in too much effort, you need more laziness in your life. ;p

Reply:
Good point, tried to abstract the underlying channel, but I agree it is a bit unnecessary.